R/load_fonts.R

```diff
@@ -8,24 +7,34 @@
 #' @export
 #'
 load_arcadia_fonts <- function(custom_font = "Suisse", fallback_font = "sans") {
-  # define font names to check for
-  font_names <- c("Suisse Int'l", "Suisse Int'l Semi Bold", "Suisse Int'l Medium", "Suisse Int'l Mono")
+  # check if custom font available
+  font_family <- custom_font
+
+  if (!("Suisse Int'l" %in% extrafont::fonts())) {
+    font_family <- system(sprintf("fc-match -f '%%{family}' %s", fallback_font), intern = TRUE)
+    # Ubuntu returns "DejaVu Sans" while R looks for "DejaVuSans", so remove spaces
+    if (Sys.info()["sysname"] == "Linux") {
+      font_family <- gsub(" ", "", font_family)
+    }
+  }
 
   # import and load fonts
-  # supress messages, below will handle incorrect loading
+  # suppress messages, below will handle incorrect loading
   suppressMessages({
-    invisible(utils::capture.output(extrafont::font_import(pattern = custom_font, prompt = FALSE)))
+    invisible(utils::capture.output(extrafont::font_import(pattern = font_family, prompt = FALSE)))
     invisible(utils::capture.output(extrafont::loadfonts(device = "pdf", quiet = TRUE)))
   })
 
-  # check if custom font available
+  # define font names to check for
+  font_names <- c("Suisse Int'l", "Suisse Int'l Semi Bold", "Suisse Int'l Medium", "Suisse Int'l Mono")
   available_fonts <- extrafont::fonts()
   missing_fonts <- setdiff(font_names, available_fonts)
+
   if (length(missing_fonts) == 0) {
     cat(sprintf("All custom fonts '%s' are successfully loaded.\n", paste(font_names, collapse = ", ")))
     return(font_names)
   } else {
     cat(sprintf("Custom fonts '%s' not found. Using fallback font '%s'.\n", paste(missing_fonts, collapse = ", "), fallback_font))
     return(rep(fallback_font, length(font_names)))
   }
-}
+}
```

Comment:
nit: missing new line EOF

Reply:
Fixed. Is there a good R formatter for vscode you can recommend?

Reply:
I don't think I know one unfortunately :( lintr is the "standard" R linter, but it isn't as superb as some of the python ones